src/common/managers/SceneManager.cpp

```diff
@@ -0,0 +1,118 @@
+/*
+** EPITECH PROJECT, 2024
+** StellarForge
+** File description:
+** SceneManager.cpp
+*/
+
+#include "SceneManager.hpp"
+
+void SceneManager::addScene(const UUID &uuid, const std::shared_ptr<IScene> &scene, int position)
+{
+    if (_scenes.find(uuid) != _scenes.end()) {
+        throw ManagerException("Scene with UUID " + uuid.getUuidString() + " already exists.");
+    }
+    _scenes[uuid] = scene;
+
+    if (position < 0 || position >= _sceneOrder.size()) {
```

Comment:
I get position can be a std::size_t to avoid negative positions

Reply:
i use -1 (default) to append at the end of the list per default, do you have something else in mind?